compiler/driver.ml

```diff
@@ -213,7 +213,7 @@ let driver source_file (options : Cli.options) : int =
         | `Typecheck ->
           Cli.debug_print "Typechecking again...";
           let _ =
-            try Shared_ast.Typing.program prgm
+            try Shared_ast.Typing.program prgm ~leave_unresolved:true
```

Comment:
I'm curious why the flag is set here ?

Reply:
You're right we're before the closure conversion so this should be set to `false`.